src/shared_modules/sync_protocol/include/metadata_provider.h

```diff
@@ -0,0 +1,126 @@
+/* Copyright (C) 2015, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef METADATA_PROVIDER_H
+#define METADATA_PROVIDER_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Agent metadata structure containing system and agent information
+ *
+ * This structure holds all metadata fields required by the sync protocol.
+ * All string fields are null-terminated C strings with fixed maximum lengths.
+ */
+typedef struct {
+    char agent_id[256];           ///< Agent identifier (e.g., "001", "000" for server)
+    char agent_name[256];         ///< Agent name
+    char agent_version[256];      ///< Wazuh agent version
+    char architecture[256];       ///< System architecture (e.g., "x86_64")
+    char hostname[256];           ///< System hostname
+    char os_name[256];                  ///< Operating system name
+    char os_type[256];                  ///< Operating system type (e.g., "linux", "windows", "darwin")
+    char os_platform[256];              ///< Operating system platform/distribution (e.g., "ubuntu", "centos", "windows")
+    char os_version[256];               ///< Operating system version
+    char os_distribution_release[256];  ///< Operating system distribution release (e.g., "1903", "sp1")
+    char os_full[256];                  ///< Operating system full version/display version (e.g., "19H1")
+    char checksum_metadata[256];        ///< Checksum of metadata fields
+    uint64_t global_version;            ///< Global version counter
```

Comment:
This shouldn't be here. It should be a parameter of `synchronizeMetadataOrGroups`.